src/cognito/index.ts

```diff
@@ -49,8 +49,29 @@ export async function withIdentityPoolId(
       transformRequest: (url: string) => {
         // Only sign Amazon Location Service URLs
         if (url.match(/^https:\/\/maps\.(geo|geo-fips)\.[a-z0-9-]+\.(amazonaws\.com)/)) {
+          const urlObj = new URL(url);
+
+          // Split the pathname into parts, using the filter(Boolean) to ignore any empty parts,
+          // since the first item will be empty because the pathname looks like:
+          //    /v2/styles/Standard/descriptor
+          const pathParts = urlObj.pathname.split("/").filter(Boolean);
+
+          // The signing service name for the standalone Maps SDK is "geo-maps"
+          let serviceName = "geo-maps";
+          if (pathParts?.[0] == "v2") {
+            // For this case, we only need to sign the map tiles, so we
+            // can return the original url if it is for descriptor, sprites, or glyphs
+            if (pathParts?.[1] != "tiles") {
+              return { url };
+            }
```

Comment:
`transformRequest` takes a `ResourceType` as its second argument (https://maplibre.org/maplibre-gl-js/docs/API/type-aliases/RequestTransformFunction/); using that will probably be slightly clearer (and less prone to URLs moving around)

Reply:
Good call, fixed!